syntax = "proto3";

package dynamos;

option go_package = "github.com/Jorrit05/DYNAMOS/pkg/proto";
import "google/protobuf/empty.proto";

// The sidecar definition.
service Generic {
    rpc InitTracer(ServiceName) returns (google.protobuf.Empty) {}
  }

message RequestMetadata {
    string correlation_id = 1;
    string destination_queue = 2;
    string job_name = 3;
    string return_address = 4;
    string job_id = 5;
    // Optional Binary or textual representation of span context (in proto3, all fields are "optional" by default (unless marked as repeated or oneof)
    // adheres to the msComm.Traces type, also expected when passing it to StartRemoteParentSpan() function
    map<string, bytes> traces = 6;
}

message ServiceName {
    string service_name = 1;
}